pkg/storage/plugins/mongodb/mongodb.go

```diff
@@ -192,58 +229,102 @@ func (s *Store) buildFindOptions(opts plugins.FindOptions) *options.FindOptions
 		query.SetSort(opts.Sort)
 	}
 
-	return query
+	return query, nil
 }
 
-func (s *Store) Insert(opts plugins.InsertOptions) error {
-	cxt, cancel := context.WithTimeout(context.Background(), s.timeout)
-	defer cancel()
+func (s *Store) Insert(ctx context.Context, opts plugins.InsertOptions) error {
+	ctx, span := tracing.StartSpan(ctx)
```

Comment:
The plugins don't strictly have to create spans, the grpc server will create them automatically. We can tweak this to either include more info or just add attributes to the parent span later.